src/infra/redis/mapper.ts

```diff
@@ -0,0 +1,74 @@
+import { RedisKey } from 'ioredis';
+import { RedisGuard, TypeGuard } from './guards/index.js';
+import { XAutoClaimResponse, XItem, XItems, XReadBufferReply } from './interfaces/redis.interface.js';
+import { StreamMessageReply, StreamsMessagesReply } from './interfaces/stream-message-reply.js';
+
+export function mapToXAutoClaimResponse(value: unknown): XAutoClaimResponse {
+	if (value === null || value === undefined) {
+		// Is this correct or should there also be an error thrown?
+		return { nextId: '', messages: null };
+	}
+
+	const unknownArray = TypeGuard.checkUnknownArrayWithElements(value);
+	const xItems = RedisGuard.checkXItems(unknownArray[1]);
+	const redisKey = TypeGuard.isString(unknownArray[0])
+		? TypeGuard.checkString(unknownArray[0])
+		: TypeGuard.checkBuffer(unknownArray[0]);
+
+	return {
+		nextId: redisKey,
+		messages: mapToStreamMessagesReplies(xItems),
+	};
+}
+
+export function mapToStreamMessagesReplies(messages: XItems | unknown): StreamMessageReply[] {
+	if (messages === null) {
```

Comment:
Wie in Zeile 7 hier auch und in Zeile 40. 
Soll bei input null einfach nichts passieren, oder soll ein Fehler fliegen, daher hier default Wert rein und dann in die check Methoden gehen.